shiny/render/transformer/_transformer.py

```diff
@@ -342,9 +344,16 @@ def _render_default(self) -> TagList | Tag | MetadataNode | str:
 
         params = tuple(inspect.signature(self.default_ui).parameters.values())
         if len(params) > 0 and params[0].name == "_params":
-            return self.default_ui(self._params.kwargs, self.__name__)  # type: ignore
+            return self.default_ui(
+                self._params.kwargs,
```

Comment:
I'm having a bit of trouble picturing this function call. Is it always true that `default_ui` can take a dictionary as the first argument?

I would have imagined that you would merge `self.default_ui_kwargs,` and `self._params.kwargs` and then pass the result in (with `**`, of course).

Reply:
The conditional just before this is checking if the first param is named `_params`, that's a signal from the `default_ui` function to us that it wants the dictionary as the first argument, and inside of that function it will then do the rewriting.

Just writing that out felt bad, so I refactored it to be much more straightforward. Thanks for the feedback